src/sequelize/models/report.js

```diff
@@ -0,0 +1,35 @@
+export default (sequelize, DataTypes) => {
+  const Report = sequelize.define('Report', {
+    id: {
+      allowNull: false,
+      autoIncrement: true,
+      primaryKey: true,
+      type: DataTypes.INTEGER
+    },
+    reporterId: {
+      type: DataTypes.INTEGER,
+      allowNull: false
+    },
+    articleId: {
+      type: DataTypes.INTEGER,
+      allowNull: false
+    },
+    slug: {
+      type: DataTypes.STRING,
+      allowNull: false
+    },
+    reason: {
+      type: DataTypes.STRING,
+      allowNull: false
+    },
+    description: {
+      type: DataTypes.STRING,
+      allowNull: false
+    }
+  });
+  Report.associate = (models) => {
+    Report.belongsTo(models.User, { foreignKey: 'reporterId', as: 'reporter' });
+    Report.belongsTo(models.Article, { foreignKey: 'articleId', as: 'reportedArticle' });
```

Comment:
Wouldn't it be good to add `onDelete: 'CASCADE'` here?

Reply:
@Malaba6 `onDelete: 'CASCADE'` should be used for the parent of the relationship i.e the one with the `has` association. Cascade means that when I delete an Article I want all it's reports deleted as well. If I delete a report, I do not want it's article also deleted. With the child onDelete 'SET NULL' or 'NO ACTION' are more appropriate options. I think it's `NO ACTION` by default.